hana/lib/HANAService.js

```diff
@@ -837,19 +841,10 @@ class HANAService extends SQLService {
         const element = elements?.[name] || {}
         // Don't apply input converters for place holders
         const converter = (sql !== '?' && element[inputConverterKey]) || (e => e)
-        let managed = element[annotation]?.['=']
-        switch (managed) {
-          case '$user.id':
-          case '$user':
-            managed = this.func({ func: 'session_context', args: [{ val: '$user.id' }] })
-            break
-          case '$now':
-            managed = this.func({ func: 'session_context', args: [{ val: '$user.now' }] })
-            break
-          default:
-            managed = undefined
-        }
-
+        const val = _managed[element[annotation]?.['=']]
+          || !isUpdate && (element['@cds.valid.from'] ? '$valid.from' : element['@cds.valid.to'] ? '$valid.to' : null)
```

Comment:
Why is `|| !isUpdate && (element['@cds.valid.from'] ? '$valid.from' : element['@cds.valid.to'] ? '$valid.to' : null)` required?

Reply:
otherwise you cannot insert/ upsert data without specifying both `validFrom` and `validAt` in the payload.
in other words:
`coalesce(ISO(value->>'$."validFrom"'), session_context('$valid.from'))`
instead of just
`ISO(value->>'$."validFrom"')`.

Reply:
@BobdenOs @johannes-vogel 
was the default `validTo` changed? i now get `$now` but i would have expected something like `9999-12-31T23:59:59.999Z`...

Reply:
`session_context('$now')` will return the timestamp from the `tx.timestamp`. Or is the `$now` part of the `entries || data` ?

Reply:
https://github.com/cap-js/cds-dbs/blob/87208af735941369a0f926873f5ddb950734b917/db-service/lib/common/session-context.js#L27 defaults to `$now` (logically)

Reply:
I see what you mean. I don't know what the behavior used to be. 
I am also not sure whether the max timestamp default is correct in the temporal context as I would expect that any new insert with the same key values should become the `validTo` of the last value.

```js
await INSERT({ID: 1/*, validFrom: $now, validTo: timestamp_max */}).into(temporal)
await sleep(...)
await Promise.all([
  INSERT({ID:1/*, validFrom: $now, validTo: timestamp_max */}).into(temporal),
  UPDATE(temporal).data({validTo: $now}).where(`ID = ${1} and validTo = ${timestamp_max}`),
])
```

Reply:
I think we did not support/document how to write temporal data so far.
For now I think it's fine that the input is required by the user → #344